Extract GPIO pin number constants to common enum

taesungh opened the pull request:
Resolves #84.

## Changes
- Move all constants for GPIO pin numbers used by components to a single module to avoid having conflicting numbers
  - `GpioPins` enum will enforce discriminant
- Use [`num_enum::IntoPrimitive`](https://docs.rs/num_enum/latest/num_enum/derive.IntoPrimitive.html) to convert from enum value to `u8` using `.into()`

taesungh committed:
Extract GPIO pin number constants to common struct

- Move all constants for GPIO pin numbers used by components to a single
  module to avoid having conflicting numbers
- Use a struct instead of enum to avoid needing to cast `as u8`

samderanova commented on pod-operation/src/utils/gpio.rs:
```diff
@@ -0,0 +1,13 @@
+
+// Not discriminant but removes need to cast
+impl GpioPins {
+	const _RESERVED_I2C_SDA: u8 = 2;
+	const _RESERVED_I2C_SCL: u8 = 3;
```
Is there a reason why you left these as private since some of our peripherals will be using these pins?

samderanova commented on pod-operation/src/utils/gpio.rs:
```diff
@@ -0,0 +1,13 @@
+#[non_exhaustive]
```
Good work pointing out that this list is currently non-exhaustive. Will we be removing this once all pins have been accounted for?

Also, is there any pro to using this versus just a comment?

samderanova commented on pod-operation/src/utils/mod.rs:
```diff
@@ -0,0 +1,2 @@
+mod gpio;
+pub use gpio::GpioPins;
```
Is it not enough to just have `mod gpio` like we have in the `mod.rs` of `components`? What is this line used for?

samderanova commented:
Thanks for working on this! Maybe there should also be some sort of mechanism that enforces only one use of each pin at a time.

I suppose we can also ignore the error in the checks regarding the unused enum field.

taesungh reviewed (commented):
A regular enum would enforce enum fields to be [discriminant](https://doc.rust-lang.org/reference/items/enumerations.html#discriminants) whereby having multiple of the same value would be an error, but using a regular enum would require casting each pin value with `as u8` when passed into `Gpio::get` which seems inconvenient (and more generally risks lossy conversion), but perhaps this is preferred to enforce discriminants.

ryescholin reviewed (approved):
Looks great!

taesungh commented:
Actually, I found a suitable compromise for having the enum, will update soon.

taesungh committed:
Use `num_enum::IntoPrimitive` for discriminant

- Install `num_enum` in `pod-operation`
- Implements `From<Enum>` for underlying u8 repr so we can use `.into()`
